Zend/zend_attributes.c

```diff
@@ -49,4 +61,10 @@ void zend_register_attribute_ce(void)
 	zend_ce_php_compiler_attribute->ce_flags |= ZEND_ACC_FINAL;
 
 	zend_compiler_attribute_register(zend_ce_php_compiler_attribute, zend_attribute_validate_phpcompilerattribute);
+
+	INIT_CLASS_ENTRY(ce, "Deprecated", NULL);
+	zend_ce_deprecated_attribute = zend_register_internal_class(&ce);
+	zend_ce_php_compiler_attribute->ce_flags |= ZEND_ACC_FINAL;
+
```

Comment:
Would it be worth to disable serialization/deserialization as well?

Reply:
No, it would not be a problem to serialize/unserilize this class, in fact if people want to cache it for whatever reason, it makes sense to allow it.